tests/Doctrine/Tests/DBAL/Schema/IndexTest.php

```diff
@@ -185,4 +185,12 @@ public function testOptions()
         self::assertSame('name IS NULL', $idx2->getOption('WHERE'));
         self::assertSame(['where' => 'name IS NULL'], $idx2->getOptions());
     }
+
+    public function testGetNameWithMultipleDots() : void
+    {
+        $name  = 'my.index.name';
```

Comment:
Please turn this into a functional test that deploys a schema containing an object that has a dot in its name and introspects it. It needs to be proven that it's practically possible and practically correct.

Reply:
In my case (an index in a mysql database which had dots in its name), the name of the index is not unique anymore when parts of the name are lost. I am not completely sure what else has to be proven here.

Reply:
> I am not completely sure what else has to be proven here.

That mysql tolerates indices with a name containing a dot, if I understood correctly

Reply:
I see (hopefully). Where should the functional test to be added (.../Functional/Schema/MysqlSchemaManagerTest.php)?

Reply:
@greg0ire, @morozov After a more testing, I realized, that this will not work in the actual way, because the namespace will be taken for the database name (as you said @morozov). What should be done here? Closing the pull request and stick with the convention that index names can't contain dots? Or refactoring it?

I still can create a functional test to show that mysql indices can have dots in the name (without dbal).

Reply:
What if the namespace contains dots too?

Reply:
After a lot of testing yesterday, I think that an index does not have a namespace. A schema, table or column can have a namespace (including dots). The name of an asset can not contain dots, except it is an index. Does this make sense?

Reply:
> But for index it is, because it does not have a namespace.

Is it true for all supported platforms? There's currently no clear separation in DBAL between indices and constraints, and depending on the platform some of them are scoped to a table and some are scoped to a schema (both of which I believe may have dots in their names if quoted).

It feels like solving this issue will require significant research first.